staging/src/k8s.io/kubectl/pkg/config/v1alpha1/types.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Preferences stores elements of KubeRC configuration file
+type Preferences struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Overrides is used to change the defaults values of flags of commands.
+	// This is especially useful, when user doesn't want to explicitly
+	// set flags each time.
+	// +listType=atomic
+	Overrides []CommandOverride `json:"overrides"`
+
+	// Aliases stores the alias definitions. If the alias name collides with
+	// a built-in command, built-in command always overrides the alias name.
+	// +listType=atomic
+	Aliases []AliasOverride `json:"aliases"`
+}
+
+// AliasOverride stores the alias definitions.
+// It is applied in a pre-defined order which is
+// kubectl [ALIAS NAME] expands to kubectl [COMMAND] [ARGUMENTS]
+type AliasOverride struct {
+	// Name is the name of alias
+	Name string `json:"name"`
+	// Command is the single or set of commands to execute, such as "set env" or "create"
+	Command string `json:"command"`
+	// Arguments is allocated for the arguments such as resource names, etc.
+	// +listType=atomic
+	Args []string `json:"args,omitempty"`
+	// Flag is allocated to store the flag definitions of alias
+	// +listType=atomic
+	Flags []CommandOverrideFlag `json:"flags,omitempty"`
+}
+
+// CommandOverride stores the commands and their associated flag's
+// default values.
+type CommandOverride struct {
+	// Command refers to a command whose flag's default value is changed.
+	Command string `json:"command"`
```

Comment:
I see this in the KEP:

```
    - command: "*"
      flags:
        - name: exec-auth-allowlist
          default: /var/kubectl/exec/...
```

what is the meaning of globs in this field? how do they match?

what is the meaning of flag overrides for flags that don't exist in the command? do they get ignored? warn? fail?

Reply:
> is this a single command (like apply) or can it take a chain of commands (like create secret)? what validation is there on this command?

This can be any command (including the command chains `create secret`). We are asserting the existence of the command first and if it doesn't exist, we short cut and fail immediately. 

> I see this in the KEP:

    - command: "*"
      flags:
        - name: exec-auth-allowlist
          default: /var/kubectl/exec/...
>what is the meaning of globs in this field? how do they match?

This functionality is not supported in this phase. But main motivation is to set persistent flags and apply for all commands. For example, set kubeconfig path with the command `*`, this will be automatically used for all commands.

> what is the meaning of flag overrides for flags that don't exist in the command? do they get ignored? warn? fail?

kuberc returns an error and this interrupts the execution with non-exit code.

Reply:
> I'm not sure there is a single flag that is supported on literally every kubectl command

Yeah, I'm also not sure this functionality will be useful anyway.

Reply:
ok, let's update the kep to drop that (or flag it as unresolved)

Reply:
This PR https://github.com/kubernetes/enhancements/pull/4925 removes this usage from KEPs.